homeassistant/components/lcn/helpers.py

```diff
@@ -247,6 +247,76 @@ def import_lcn_config(lcn_config: ConfigType) -> list[ConfigType]:
     return list(data.values())
 
 
+def purge_entity_registry(
+    hass: HomeAssistant, entry_id: str, entry_data: ConfigType
+) -> None:
+    """Remove orphans from entity registry which are not in entry data."""
+    entity_registry = er.async_get(hass)
+
+    # Find all entities that are referenced in the config entry.
+    references_config_entry = {
+        entity.entity_id
+        for entity in entity_registry.entities.values()
```

Comment:
We have this helper:
https://github.com/home-assistant/core/blob/72dbca4f5bc3a40ac96b2c8fc355ce85bedd003c/homeassistant/helpers/entity_registry.py#L851